### PR TITLE
Add BuildScript::out_dir

### PR DESCRIPTION
**Pull request by ehuss**
Added in https://github.com/rust-lang/cargo/pull/7622.

### CODE REVIEW

File: src/messages.rs [hunk 1]
```diff
@@ -75,6 +75,11 @@ pub struct BuildScript {
     pub cfgs: Vec<PathBuf>,
     /// The environment variables to add to the compilation
     pub env: Vec<(String, String)>,
+    /// The `OUT_DIR` environment variable where this script places its output
+    ///
+    /// Added in Rust 1.41.
+    #[serde(default)]
+    pub out_dir: String,
```
Comment (oli-obk) [inline src/messages.rs]: This could be a `PathBuf`, right?

Comment (ehuss) [reply src/messages.rs]: You're right, good catch.

File: src/messages.rs [hunk 2]
```diff
@@ -75,6 +75,11 @@ pub struct BuildScript {
     pub cfgs: Vec<PathBuf>,
     /// The environment variables to add to the compilation
     pub env: Vec<(String, String)>,
+    /// The `OUT_DIR` environment variable where this script places its output
+    ///
+    /// Added in Rust 1.41.
+    #[serde(default)]
```
Comment (matklad) [inline src/messages.rs:81]: This caused a runtime panic in rust-analyzer(nothing to problematic, just a minor annoyance :) ) I guess this is an argument for using options in similar cases in the future.

https://github.com/rust-analyzer/rust-analyzer/issues/5125

Comment (oli-obk) [reply src/messages.rs:81]: oh.. I guess just using `default` for older compilers isn't making this backwards compatible. Thanks, I'll keep that in mind for the future.

**Pull request merged**